plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/selectionwizard/NewMenuWizard.java

```diff
@@ -11,37 +11,85 @@
 
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.common.wait.WaitWhile;
 import org.eclipse.reddeer.eclipse.ui.dialogs.NewWizard;
 import org.eclipse.reddeer.jface.window.Openable;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+import org.eclipse.reddeer.swt.impl.button.FinishButton;
+import org.eclipse.reddeer.swt.impl.button.NoButton;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
 
 /**
- * Represents wizard which can be found in new wizard dialog (menu 'File -&gt; New -&gt; Other...').
+ * Represents wizard which can be found in new wizard dialog (menu 'File -&gt;
+ * New -&gt; Other...').
+ * 
  * @author rawagner
+ * @contributor jkopriva@redhat.com
  *
  */
-public abstract class NewMenuWizard extends AbstractSelectionWizardDialog{
-	
+public abstract class NewMenuWizard extends AbstractSelectionWizardDialog {
+
 	/**
 	 * Constructs new Wizard that can be found in NewWizard.
-	 * @param shellText new wizard text when next is clicked (ie 'New File')
+	 * 
+	 * @param shellText      new wizard text when next is clicked (ie 'New File')
 	 * @param wizardCategory new wizard category (ie 'General')
-	 * @param wizardName new wizard name (ie 'File')
+	 * @param wizardName     new wizard name (ie 'File')
 	 */
 	public NewMenuWizard(String shellText, String wizardCategory, String wizardName) {
 		super(shellText, wizardCategory, wizardName);
 	}
-	
+
 	/**
 	 * Constructs new Wizard that can be found in NewWizard.
-	 * @param shellText new wizard text when next is clicked (ie 'New File')
+	 * 
+	 * @param shellText  new wizard text when next is clicked (ie 'New File')
 	 * @param wizardPath wizard path in new wizard (ie 'General, File')
 	 */
 	public NewMenuWizard(String shellText, String[] wizardPath) {
-		super(shellText,wizardPath);
+		super(shellText, wizardPath);
 	}
-	
+
 	@Override
 	protected Openable getOpenAction() {
 		return new SelectionWizardOpenable(new NewWizard(), wizardPath, matcher);
 	}
+
+	/**
+	 * Click the finish button in wizard dialog.
+	 * 
+	 * @param openPerspective open perspective related to the new project
+	 */
+	public void finish(boolean openPerspective) {
+		finish(TimePeriod.LONG, openPerspective);
+	}
+
+	/**
+	 * Click the finish button in wizard dialog.
+	 * 
+	 * @param timePeriod timeout for running job after clicking on Finish button 
+	 * @param openPerspective open perspective related to the new project
+	 */
+	public void finish(TimePeriod timePeriod, boolean openPerspective) {
+		checkShell();
+		log.info("Finish wizard");
+
+		new FinishButton(this).click();
+
+		try {
+			new WaitWhile(new JobIsRunning(), timePeriod);
+		} catch (NoClassDefFoundError e) {
+			// do nothing, reddeer.workbench plugin is not available
+		}
+		new WaitUntil(new ShellIsAvailable("Open Associated Perspective?"));
```

Comment:
There are hundres of test failures at prcheck job: https://hudson.eclipse.org/reddeer/job/reddeer.prcheck. This is caused by the fact that not all new menu wizard dialogs are offering Open Perspective option after finish is pressed. I would go one of these ways:
1. If user chooses not to open perspective, then check for the existence of dialog with ShellIsAvailable.getResult and then decide if just click No button if dialog was thrown, do nothing otherwise. OR If user chooses to open the perspective and the dialog is not at place throw an exception, open it otherwise.
2. Find out if menu wizard supports opening the perspective and make it a condition

Reply:
Hm, this is quite strange as you did not overrided the finish method but added new taking boolean parameter. This stack trace from failing test should not happen:
`org.eclipse.reddeer.common.exception.WaitTimeoutExpiredException: Timeout after: 10 s.: shell matching Matcher matching when all matchers match: [Matcher matching widget which text matches: "Open Associated Perspective?"] not found.
	at org.eclipse.reddeer.common.wait.WaitUntil.throwWaitTimeOutException(WaitUntil.java:122)
	at org.eclipse.reddeer.common.wait.AbstractWait.timeoutExceeded(AbstractWait.java:207)
	at org.eclipse.reddeer.common.wait.AbstractWait.wait(AbstractWait.java:151)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:132)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:97)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:64)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:50)
	at org.eclipse.reddeer.common.wait.WaitUntil.<init>(WaitUntil.java:38)
	at org.eclipse.reddeer.eclipse.selectionwizard.NewMenuWizard.finish(NewMenuWizard.java:100)
	at org.eclipse.reddeer.eclipse.selectionwizard.NewMenuWizard.finish(NewMenuWizard.java:71)
	at org.eclipse.reddeer.eclipse.test.core.resources.MavenProjectTest.createProject(MavenProjectTest.java:46)`